client.go

```diff
@@ -542,6 +542,35 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 	return client, nil
 }
 
+// pgIdentifierRegexp matches all valid PostgreSQL identifier names:
+var pgIdentifierRegexp = regexp.MustCompile("^[a-zA-Z_][a-zA-Z0-9_]*$")
+
+// Listen subscribes to a notification topic. The notifyFunc will be called for
+// any notification that is received on the topic. The returned subscription must be
+// closed with Unlisten when no longer needed.
+//
+// The provided notifyFunc should be fast and not block to avoid dropping other
+// notifications and potentially causing internal issues within River.
+//
+// It returns an error if the specified topic is not a valid Postgres
+// identifier. Panics if notifyFunc is nil.
+func (c *Client[TTx]) Listen(topic string, notifyFunc NotifyFunc) (*ListenSubscription, error) {
+	// Validate that the topic is a valid Postgres identifier:
```

Comment:
@brandur should we also forbid the use of `river_` prefix topics? Or do you think there's no real risk in allowing it? I've gone back and forth but atm I'm not concerned.

Reply:
Hmm, doesn't hurt to start with constraints first and relax them later if it seems necessary.